implementation/src/main/java/io/smallrye/metrics/exporters/PrometheusExporter.java

```diff
@@ -354,19 +359,22 @@ private void createSimpleValueLine(StringBuffer sb, MetricRegistry.Type scope, S
     }
 
 
-    private String getPrometheusMetricName(Metadata entry, String name) {
+    static String getPrometheusMetricName(String name) {
         String out = name.replaceAll("[^\\w]+",USCORE);
         out = decamelize(out);
-        if (entry == null) {
-            throw new IllegalStateException("No entry for " + name + " found");
-        }
         out = out.replace("__", USCORE);
         out = out.replace(":_", ":");
 
         return out;
     }
 
-    private String decamelize(String in) {
-        return in.replaceAll("(.)(\\p{Upper})", "$1_$2").toLowerCase();
+    private static String decamelize(String in) {
+        Matcher m = Pattern.compile("(?<=[a-z])[A-Z]").matcher(in);
```

Comment:
Shouldn't that be created in the constructor, as this method is called quite often?

Reply:
@pilhuhn I have amended the patch